spaceship/lib/spaceship/connect_api/client.rb

```diff
@@ -0,0 +1,190 @@
+require_relative '../client'
+require_relative './response'
+
+module Spaceship
+  class ConnectAPI
+    class Client < Spaceship::Client
+      attr_accessor :token
+
+      #####################################################
+      # @!group Client Init
+      #####################################################
+
+      # Instantiates a client with a JWT token.
+      def initialize(cookie: nil, current_team_id: nil, token: nil)
+        if token.nil?
+          super(cookie: cookie, current_team_id: current_team_id)
+        else
```

Comment:
Perhaps we should extract jwt auth initialization into a class-method, and the have the initializer delegate to the class-method or `super`

Reply:
I'm not sure if moving this to a class method would make things much cleaner 🤔 We are setting 3 instance variables here that would feel a bit weird to set outside of an initializer unless I'm picturing this wrong.

Reply:
I was imagining something like:
```ruby
def self.token_auth(token)
   connect_client = self.new(current_team_id: current_team_id)
   connect_client.client = Faraday.new( ... )
   return connect_client
end

def initialize( token:, current_team_id:, cookie: )
  if token
    self.class.token_auth(token)
  else
    super( ... )
  end
end
```

I made the assumption that all of the instance variables set during initialization have public accessors. That seems correct, tho.